### PR TITLE
Configure checkstyle, remove code-assert

### LINKED ISSUE #1797: Replace code-assert with alternative checkstyle test
> I also think we should get rid of `guru.nidi:code-assert` alltogether as it has a humongous dependency tree and it's only used in one testcase:
> 
> _Originally posted by @mprins in https://github.com/oshi/oshi/issues/1796#issuecomment-979039947_

The one testcase code-assert is used in is completely intended to give a runtime test enforcement of checkstyle issues.   Prior to using this, we were only finding these issues in the Appveyor builds, and when I say "finding" I mean failing builds and spending 10 minutes poring over the log to try to figure out the issue, instead of having it print out nicely.

I'm okay with removing code-assert if we properly configure the checkstyle plugin to fail the build on warnings, see [here, for example](https://stackoverflow.com/questions/52006084/maven-checkstyle-plugin-doesnt-fail-during-build-even-though-failsonerror-is/52006315).  Other potential options are using one of our existing CI integrations that integrates with PRs, or a Github Action that tests checkstyle, like [this one](https://github.com/jwgmeligmeyling/checkstyle-github-action)

### LINKED ISSUE #1799: Block imports not on module path
We want to avoid importing classes to the project from any packages not included as top-level dependencies (which are also included in the module descriptor for the java11 branch).  This aligns with the Java Modules need for an explicit `requires transitive` statement from these dependencies, and helps keep surprises with java11 releases to a minimum.

Consider the below maven plugin if it's easy to use, or find another way to block imports that aren't explicitly listed as dependencies in the pom, or specifically included (with associated `requires transitive` statement).

> Use this -> https://github.com/skuzzle/restrict-imports-enforcer-rule, works well to block the import usage.  You could also block junit4 as well as noted.
> 
> _Originally posted by @hazendaz in https://github.com/oshi/oshi/issues/1794#issuecomment-978969292_

### PR DESCRIPTION
**Pull request by dbwiddis**
Fixes #1797 
Fixes #1799

### CODE REVIEW

File: oshi-core/src/main/java/oshi/util/platform/windows/PerfCounterQuery.java [hunk 1]
```diff
@@ -92,14 +92,14 @@ private PerfCounterQuery() {
      */
     public static <T extends Enum<T>> Map<T, Long> queryValues(Class<T> propertyEnum, String perfObject,
             String perfWmiClass) {
-        if (!failedQueryCache.contains(perfObject)) {
+        if (!FAILED_QUERY_CACHE.contains(perfObject)) {
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/platform/windows/PerfCounterQuery.java:95]: @sonatype-lift ignore

The container is a synchronized set, you're just dumb.

File: oshi-core/src/main/java/oshi/util/platform/windows/PerfCounterQuery.java [hunk 2]
```diff
@@ -92,14 +92,14 @@ private PerfCounterQuery() {
...
             Map<T, Long> valueMap = queryValuesFromPDH(propertyEnum, perfObject);
             if (!valueMap.isEmpty()) {
                 return valueMap;
             }
             // If we are here, query failed
             LOG.warn("Disabling further attempts to query {}.", perfObject);
-            failedQueryCache.add(perfObject);
+            FAILED_QUERY_CACHE.add(perfObject);
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/platform/windows/PerfCounterQuery.java:102]: @sonatype-lift ignore

The container is a synchronized set, you're just dumb.

File: oshi-core/src/main/java/oshi/util/platform/windows/PerfCounterWildcardQuery.java [hunk 3]
```diff
@@ -83,15 +83,15 @@ private PerfCounterWildcardQuery() {
      */
     public static <T extends Enum<T>> Pair<List<String>, Map<T, List<Long>>> queryInstancesAndValues(
             Class<T> propertyEnum, String perfObject, String perfWmiClass) {
-        if (!failedQueryCache.contains(perfObject)) {
+        if (!FAILED_QUERY_CACHE.contains(perfObject)) {
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/platform/windows/PerfCounterWildcardQuery.java:86]: @sonatype-lift ignore

The container is a synchronized set, you're just dumb.

File: oshi-core/src/main/java/oshi/util/platform/windows/PerfCounterWildcardQuery.java [hunk 4]
```diff
@@ -83,15 +83,15 @@ private PerfCounterWildcardQuery() {
...
             Pair<List<String>, Map<T, List<Long>>> instancesAndValuesMap = queryInstancesAndValuesFromPDH(propertyEnum,
                     perfObject);
             if (!instancesAndValuesMap.getA().isEmpty()) {
                 return instancesAndValuesMap;
             }
             // If we are here, query failed
             LOG.warn("Disabling further attempts to query {}.", perfObject);
-            failedQueryCache.add(perfObject);
+            FAILED_QUERY_CACHE.add(perfObject);
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/platform/windows/PerfCounterWildcardQuery.java:94]: @sonatype-lift ignore

The container is a synchronized set, you're just dumb.

**Review by hazendaz — APPROVED** [review]

**Comment by dbwiddis** [comment]
Well, that took way too long for a chore, but was a nice 6 hours of learning experience.

**Review by mprins — APPROVED** [review]

**Pull request merged**